### PR TITLE
[KYUUBI #4412][FOLLOWUP] Fallback to new engine session handle for UT

### PR DESCRIPTION
**Pull request by turboFei**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala [hunk 1]
```diff
@@ -183,7 +184,8 @@ class KyuubiSyncThriftClient private (
     engineAliveProbeClient.foreach { aliveProbeClient =>
       val sessionName = SessionHandle.apply(_remoteSessionHandle).identifier + "_aliveness_probe"
       Utils.tryLogNonFatalError {
-        req.setConfiguration((configs ++ Map(KyuubiConf.SESSION_NAME.key -> sessionName)).asJava)
+        req.setConfiguration((configs ++ Map(KyuubiConf.SESSION_NAME.key -> sessionName) ++ Map(
+          KYUUBI_SESSION_HANDLE_KEY -> UUID.randomUUID().toString)).asJava)
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala:188]: use different session handle for alive probe

File: kyuubi-server/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerUserSuite.scala [hunk 2]
```diff
@@ -366,11 +366,14 @@ class KyuubiOperationPerUserSuite
   }
 
   test("align the server session handle and engine session handle for Spark engine") {
-    withJdbcStatement() { _ =>
-      val session =
-        server.backendService.sessionManager.allSessions().head.asInstanceOf[KyuubiSessionImpl]
-      val engineSessionHandle = SessionHandle.apply(session.client.remoteSessionHandle)
-      assert(session.handle === engineSessionHandle)
+    withSessionConf(Map(
+      KyuubiConf.SESSION_ENGINE_LAUNCH_ASYNC.key -> "false"))(Map.empty)(Map.empty) {
```
Comment (turboFei) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerUserSuite.scala:370]: launch async false

**Comment by turboFei** [comment]
@pan3793  fixed

**Comment by turboFei** [comment]
merging to master, because it is a minor pr and block the main branch.

**Comment by pan3793** [comment]
Thanks, late LGTM

**Pull request closed without merging**